lib/Storage.js

```diff
@@ -1,87 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
 const {Cu} = require('chrome');
-var {indexedDB} = require('sdk/indexed-db');
+Cu.import('resource://gre/modules/XPCOMUtils.jsm'); /* global XPCOMUtils */
 
-exports.Storage = function(prefix, sandbox=null) {
-  if (sandbox === null) {
-    sandbox = {Promise};
-  }
+XPCOMUtils.defineLazyModuleGetter(this, 'JSONFile',
+                                  'resource://gre/modules/JSONFile.jsm'); /* global JSONFile */
 
-  const idbPromise = new sandbox.Promise((resolve, reject) => {
-    let request = indexedDB.open('normandy', 1);
-    request.onsuccess = event => resolve(event.target.result);
-    request.onerror = err => reject(err);
+XPCOMUtils.defineLazyModuleGetter(this, 'OS',
+                                  'resource://gre/modules/osfile.jsm'); /* global OS */
+XPCOMUtils.defineLazyModuleGetter(this, 'Task',
+                                  'resource://gre/modules/Task.jsm'); /* global Task */
 
-    request.onupgradeneeded = event => {
-      var db = event.target.result;
-      db.createObjectStore('storage', {keyPath: 'key'});
-    };
-  });
+const {Log} = require('./Log.js');
+
+let storePromise;
+
+function loadStorage() {
+  if (storePromise === undefined) {
+    let path = OS.Path.join(OS.Constants.Path.profileDir, 'recipe-client.json');
```

Comment:
@gijsk Got any feedback on the filename choice here? Are there practices around how we stuff data into the profile folder?